server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -400,18 +390,19 @@ private static void validateDynamicTemplate(Mapper.TypeParser.ParserContext pars
             return;
         }
 
-        final XContentFieldType[] types;
-        if (template.getXContentFieldType() != null) {
-            types = new XContentFieldType[]{template.getXContentFieldType()};
-        } else if (template.isRuntimeMapping()) {
-            types = Arrays.stream(XContentFieldType.values()).filter(XContentFieldType::supportsRuntimeField)
-                .toArray(XContentFieldType[]::new);
-        } else {
-            types = XContentFieldType.values();
+        if (parserContext.indexVersionCreated().before(Version.V_8_0_0)) {
+            if (template.pathMatch() == null && template.match() == null && template.getXContentFieldTypes().length == 0) {
+                throw new MapperParsingException("template [" + template.name() +
+                    " does not have [match] not [path_match] or [match_mapping_type] defined");
```

Comment:
does not have [match] nor [path_match] nor [match_mapping_type] defined ?

Reply:
Good catch, fixed in https://github.com/elastic/elasticsearch/pull/69948/commits/8c9e4020ecf2bf37cd20039729ccacb73e457a9f